lib/rucio/tests/test_credential.py

```diff
@@ -0,0 +1,121 @@
+                     '0Db%2BQ4a0wICQ2PZzUfTuHXQF8dUbMJG04VH90U5EMzYg3qSUGyfnp6Jptnvgivf7iSHepJsYhyAYSBGs'
+                     'bvTOqf%2BXMQHR5VTh06G8WriZPV2OgSJ61c8qY7k8h0ju4bwcdDMFD2CT933KsnYSVatLN3EfORonLLZv'
+                     'Ydgf0WCQjUcVKRv8zY65HJS6ZKoCjhOqNBJNlpI6uR54MhmLN2CJWch1MnLIdO6bKfDup%2Bzkt8e9Xe9S'
+                     '8pTeva5cN8ZFlMkeCz7JvNkVJb1KPhI1XHPWyfuPUa2ALHh9wAD2yFSOU3cDiORFE6A%3D%3D')
+
+        assert_equal(get_signed_url('gcs', 'write', 'http://storage/directory/file', lifetime=None),
+                     'https://storage.googleapis.com/directory/file?GoogleAccessId=rucio-test@rucio-test'
+                     '.iam.gserviceaccount.com&Expires=0&Signature=Gn%2FL0%2FjGkBIdpHZ9bKw7tvqRCdslC11gt'
+                     'jbLk5AG2jA4Ywd6mTvOinUB%2BZxHY2I3XzEuMfyMnFj0vfXSemN6XmmcQkiQBhl6P3zr0GrOuO4y0xjKT'
+                     'am1MijMKLKFS9pZ6BBYrFgwKcYUcGJmVpq0Fo%2Bl5pLovBKhJbi3RE0YbGTCDA5UEM6WuWLMcQiY8smfK'
+                     '6EH9bW5tAEs70vOwNNPPUm%2FbcNKnR4z6jqThXw2mn375L02SRPx1qQ853sZKHng6O4ydm%2BSW8i7rb1'
+                     '%2BnqImWDOdvmcLIZzc6x9l6b7ETOqSL2OqOCStpBHPzpQU0spgJS96IB09uGRQum1Ej2ui5g%3D%3D')
+
+        assert_equal(get_signed_url('gcs', 'delete', 'http://storage/directory/file', lifetime=None),
+                     'https://storage.googleapis.com/directory/file?GoogleAccessId=rucio-test@rucio-test'
+                     '.iam.gserviceaccount.com&Expires=0&Signature=FVDNroX1epdTCv%2BC74o%2B8uWyvJXrqiIWg'
+                     'kdcedaOoryhRMjuv%2FVdKecnhViY%2BGOP%2B0CoI1uFOHBz%2B%2Bm10U9A3i%2B1v7AZRN5L6nbbS%2'
+                     'BJTk4oiSBMJ3FpNT9knbOVd4aSPdiBwfTybwpkWSzEb8cKQsqzrGZk4hVffipMOKkxj7UgMe%2F0DiwqyF'
+                     'o3NZsey12b9TG2xPVCZ5mJdIvJY0E5KiqEGXVCVChEhecZEyP0cUxjs8xM%2BxhOJ%2BioPQzRsFwVKtVv'
+                     'LXestniEGBMY8SY4UuthQVO1Kmq2hg30KcsgXpLzAFheK1tz0GunqPU7%2BYACZMuHj1Hp%2BTnvKNxVuJ'
+                     '5MT5g%3D%3D')
+
+    def test_list_replicas_sign_url(self):
+        """ CREDENTIAL: List replicas for an RSE where signature is enabled """
+
+        replicas = [r for r in self.rc.list_replicas(dids=[{'scope': 'mock',
+                                                            'name': f['name'],
+                                                            'type': 'FILE'} for f in self.files],
+                                                     rse_expression=self.rse)]
+        found_pfns = [replica['pfns'].keys()[0] for replica in replicas]
+        expected_pfns = ['https://storage.googleapis.com:443/atlas-europe-west1/mock/04/92/file-on-gcs_0',
+                         'https://storage.googleapis.com:443/atlas-europe-west1/mock/c6/5f/file-on-gcs_1',
+                         'https://storage.googleapis.com:443/atlas-europe-west1/mock/03/eb/file-on-gcs_2']
+        assert_equal(sorted(found_pfns), sorted(expected_pfns))
+
+        add_rse_attribute(rse=self.rse, key='sign_url_gcs', value=True)
```

Comment:
`sign_urls` is more generic as it should be extend to s3, swift, etc

Reply:
fixed